src/lib.rs

```diff
@@ -45,10 +48,22 @@
 #[cfg(test)] #[macro_use] extern crate log;
 #[cfg(all(windows, test))] extern crate advapi32;
 
-use std::cmp::Ordering;
-use std::error::Error;
-use std::fmt;
-use std::ops::{Add, Sub};
+#[cfg(feature = "std")]
+mod std_impl;
+
+#[cfg(feature = "nightly")]
```

Comment:
I understand your concern, but without this feature there's no way to do `strptime` or similar in no-std. The necessary structs are only available via alloc/nightly